app/desktop/studio_server/provider_api.py

```diff
@@ -77,6 +82,69 @@ async def connect_ollama(custom_ollama_url: str | None = None) -> OllamaConnecti
     return ollama_connection
 
 
+async def connect_docker_model_runner(
+    custom_docker_url: str | None = None,
+) -> DockerModelRunnerConnection:
+    if (
+        custom_docker_url
+        and not custom_docker_url.startswith("http://")
+        and not custom_docker_url.startswith("https://")
+    ):
+        raise HTTPException(
+            status_code=400,
+            detail="Invalid Docker Model Runner URL. It must start with http:// or https://",
+        )
+
+    try:
+        from kiln_ai.adapters.docker_model_runner_tools import (
```

Comment:
nit: import should be at the top of the file unless there reasons it needs to have it's loading delayed